app/components/ConnectDapp/ConnectionUrlForm/ConnectionUrlForm.jsx

```diff
@@ -15,16 +15,16 @@ type Props = {
 
 const ConnectionUrlForm = ({ onURI }: Props) => {
   const [url, setUrl] = useState('')
-  const [buttonDisabled, setButtonDisabled] = useState(false)
+  const buttonDisabled = useRef(false)
```

Comment:
What is the point of this change? seems more simple to use the `useState` hook

Reply:
The point is performance, this variable is just to control not executing the function more than once at a time. The useState hook causes a rerender in the component and there is no reason to do that, since it has no visual effect for the disabled button, while the useRef hook does not rerender and serves only to control the execution of the function.

Reply:
"there is no reason to do that, since it has no visual effect for the disabled button" it does have a reason, the button in the UI should be disabled if the form is invalid 😅 which is a different visual state which means we do want a rerender... I dont even see it anymore in the refactored component 😢

Reply:
I agree, it should have a state of disabled in the ui, but before my changes the button doesn't disable, so I didn't change that, since my issue was just to change the functionality and not the design.